packages/core/src/anchor/ethereum/ethereum-anchor-validator.ts

```diff
@@ -137,32 +161,64 @@ export class EthereumAnchorValidator implements AnchorValidator {
-
     const [transaction, block] = await this._getTransactionAndBlockInfo(anchorProof.chainId, txHash)
     const txValueHexNumber = parseInt(transaction.data, 16)
     const rootValueHexNumber = parseInt('0x' + anchorProof.root.toString(base16), 16)
+    return new this.ValidationResult(transaction, block, txValueHexNumber, rootValueHexNumber)
+  }
 
-    if (txValueHexNumber !== rootValueHexNumber) {
+  /**
+   * Validate version 1 anchor proof on the chain by parsing first encoded parameter
+   * @param anchorProof - Anchor proof instance
+   */
+  async validateContract(anchorProof: AnchorProof): Promise<ValidationResult> {
+    const decoded = decode(anchorProof.txHash.multihash.bytes)
+    const txHash = '0x' + uint8arrays.toString(decoded.digest, 'base16')
+    const [transaction, block] = await this._getTransactionAndBlockInfo(anchorProof.chainId, txHash)
+    const decodedArgs = iface.decodeFunctionData('anchor', transaction.data)
+    const rootCID = decodedArgs[0]
+    const txValueHexNumber = parseInt(rootCID, 16)
+    const rootValueHexNumber = parseInt('0x' + anchorProof.root.toString(base16), 16)  
+    return new this.ValidationResult(transaction, block, txValueHexNumber, rootValueHexNumber)
+  }
+  ValidationResult
+  async validate(anchorProof: AnchorProof): Promise<ValidationResult> {
+    if(anchorProof.version === 1){
+      return this.validateContract(anchorProof)
+    }else{
+      return this.validateLegacy(anchorProof)
+    }
+  }
+  
+  async validateChainInclusion(anchorProof: AnchorProof): Promise<void> {
+    const validationResult: ValidationResult = await this.validate(anchorProof)
+    if (validationResult.txValueHexNumber !== validationResult.rootValueHexNumber) {
       throw new Error(`The root CID ${anchorProof.root.toString()} is not in the transaction`)
     }
 
-    if (anchorProof.blockNumber !== transaction.blockNumber) {
+    if (anchorProof.blockNumber !== validationResult.txResponse.blockNumber) {
       throw new Error(
-        `Block numbers are not the same. AnchorProof blockNumber: ${anchorProof.blockNumber}, eth txn blockNumber: ${transaction.blockNumber}`
+        `Block numbers are not the same. AnchorProof blockNumber: ${anchorProof.blockNumber}, eth txn blockNumber: ${validationResult.txResponse.blockNumber}`
       )
     }
 
-    if (anchorProof.blockTimestamp !== block.timestamp) {
+    if (anchorProof.blockTimestamp !== validationResult.block.timestamp) {
       throw new Error(
-        `Block timestamps are not the same. AnchorProof blockTimestamp: ${anchorProof.blockTimestamp}, eth txn blockTimestamp: ${block.timestamp}`
+        `Block timestamps are not the same. AnchorProof blockTimestamp: ${anchorProof.blockTimestamp}, eth txn blockTimestamp: ${validationResult.block.timestamp}`
       )
     }
+
+    const BLOCK_THRESHHOLD = 1 //1000000000; //put into config, check where
+    if (validationResult.txResponse.blockNumber < BLOCK_THRESHHOLD) {
```

Comment:
This checks if the block number is less than the threshold. This should be checking if the block  number is greater than the threshold and the version is 0 or non existent.